core/src/main/java/org/axonframework/commandhandling/AggregateAnnotationCommandHandler.java

```diff
@@ -219,4 +175,170 @@ public Object handle(CommandMessage<?> command) throws Exception {
+         * {@link AnnotationCommandHandlerAdapter} will handle. Either this field or the {@link #aggregateType(Class)}
+         * should be provided to correctly instantiate an {@link AggregateAnnotationCommandHandler}.
+         *
+         * @param aggregateModel the {@link AggregateModel} of generic type {@code T} of the aggregate this
+         *                       {@link Repository} will store
+         * @return the current Builder instance, for fluent interfacing
+         */
+        public Builder<T> aggregateModel(AggregateModel<T> aggregateModel) {
+            assertNonNull(aggregateModel, "AggregateModel may not be null");
+            this.aggregateModel = aggregateModel;
+            return this;
+        }
+
+        /**
+         * Instantiate the {@link AggregateModel} of generic type {@code T} describing the structure of the Aggregate
+         * this {@link AggregateAnnotationCommandHandler} will handle commands for.
+         *
+         * @return a {@link AggregateModel} of generic type {@code T} describing the Aggregate this {@link
+         * AggregateAnnotationCommandHandler} will handle commands for
+         */
+        private AggregateModel<T> buildAggregateModel() {
+            if (aggregateModel == null) {
+                return inspectAggregateModel();
+            } else {
+                return aggregateModel;
+            }
+        }
+
+        private AggregateModel<T> inspectAggregateModel() {
+            if (parameterResolverFactory == null) {
+                parameterResolverFactory = ClasspathParameterResolverFactory.forClass(aggregateType);
+            }
+
+            return handlerDefinition == null
+                    ? AnnotatedAggregateMetaModelFactory.inspectAggregate(aggregateType, parameterResolverFactory)
+                    : AnnotatedAggregateMetaModelFactory.inspectAggregate(aggregateType,
+                                                                          parameterResolverFactory,
+                                                                          handlerDefinition);
+        }
+
+        /**
+         * Initializes a {@link AggregateAnnotationCommandHandler} as specified through this Builder.
+         *
+         * @return a {@link AggregateAnnotationCommandHandler} as specified through this Builder
+         */
+        public AggregateAnnotationCommandHandler<T> build() {
+            return new AggregateAnnotationCommandHandler<>(this);
+        }
+
+        /**
+         * Validate whether the fields contained in this Builder are set accordingly.
+         *
+         * @throws AxonConfigurationException if one field is asserted to be incorrect according to the Builder's
+         *                                    specifications
+         */
+        protected void validate() throws AxonConfigurationException {
+            assertNonNull(repository, "The Repository is a hard requirement and should be provided");
```

Comment:
`Repository` cannot be `null`.

Reply:
Well, the `Repository` does not have a default, hence should be provided. Thus, the `build()` function can be called without ever having called `repository()` prior to that.
Hence I think this assertion is still justified.